### DISCUSSION TITLE
Welcome to koka Discussions

### DISCUSSION THREAD

**Discussion by daanx** [koka-lang/koka#109] [General]
## Welcome

We are using Discussions as a place to connect with other members of our community without needing to open an issue.

* Ask questions you are wondering about,
* Share ideas,
* Engage with other community members,
* Welcome others and be open-minded. Remember that this is a community we build together.

**Comment by zakkor** [comment]
@daanx Hey there! Are there any real-time chat communities for Koka (like IRC, Discord, or something else)?

**Reply by zakkor** [reply]
Ah, there is a gitter room (completely missed the badge on the Koka website): https://gitter.im/koka-lang/koka

**Comment by KilianVounckx** [comment]
Hiya,

Loving the language so far! It's like the perfect mix of functional programming with imperative elements.

I was wondering about something. In the book, it is stated that koka is the [first language with trailing lambda's](https://koka-lang.github.io/koka/doc/book.html#sec-with). I was wondering how this relates to [julia](https://julialang.org)'s [do-block syntax for function arguments](https://docs.julialang.org/en/v1/manual/functions/#Do-Block-Syntax-for-Function-Arguments). It seems very similar, but do you know if there are any differences in the inner implementation?

I totally get it if you don't know about it. No one can know all languages.

**Reply by anfelor** [reply]
Ah, that looks cool! I think one difference is that Julia inserts the function as the first argument whereas Koka inserts it as the last.
In Koka, you can also use the dot-syntax for the first argument:
```koka
[a,b,c].map fn(x)
    if x < 0 && iseven(x) then 0
    elif x == 0 then 1
    else x
```
